### DISCUSSION TITLE
Exit button in Scrollable Jump Table (static)

### DISCUSSION THREAD

**Discussion by JosehL7** [olikraus/u8g2#2516] [Q&A] [answered]
Hello, I'm starting my studies with the library and I saw the example of a scrollable jump table (static) as below:

```
MUI_FORM(1)
MUI_STYLE(0)
MUI_DATA("GP", 
    MUI_10 "Goto Buttons|"
    MUI_20 "uint8 Number|"
    MUI_30 "uint8 Checkbox|"
    MUI_40 "uint8 Cycle Options|"
    MUI_50 "uint8 ParentChild Select|"
    MUI_60 "uint8 Char/Text Input|"
    MUI_70 "uint16 Cycle Options|"
    MUI_80 "uint16 ParentChild Select")
MUI_XYA("GC", 5, 25, 0) 
MUI_XYA("GC", 5, 37, 1) 
MUI_XYA("GC", 5, 49, 2) 
MUI_XYA("GC", 5, 61, 3) 
```

From what I understood, when I select each of these fields I can go to another form. I would like to know if one of the items in this list could actually be a back button so I could exit the menu. I tried using the "Exit button" but it remains static on the display and I would actually like it to be just another item in the list.

I appreciate any help in advance.

**Comment by olikraus** [comment]
Let me assume you use `uint8_t mui_u8g2_goto_form_w1_pi(mui_t *ui, uint8_t msg)`, then we could create a new muif function based on uint8_t mui_u8g2_goto_form_w1_pi(mui_t *ui, uint8_t msg):

``` C++
uint8_t mui_goto_or_back_form_w1_pi(mui_t *ui, uint8_t msg)
{
  if ( msg == MUIF_MSG_CURSOR_SELECT )
  {
    if ( mui_GetSelectableFieldTextOption(ui, ui->last_form_fds, ui->arg + ui->form_scroll_top) )
    {
      if ( (uint8_t)ui->text[0] == 255 ) // use MUI_255 for the return
      {
         return mui_RestoreForm(ui);
      }
    }
  }
  return mui_u8g2_goto_form_w1_pi(ui,  msg);  // call the original MUIF
}
```

In this case, the jump back functionality is then executed with MUI_255.
Not tested...

**Comment by JosehL7** [comment]
Hello Olikraus

I tested the code you sent, but it was showing an error because the "`mui_RestoreForm(ui);`" function is of the "void" type and I can't return it in the "`mui_goto_or_back_form_w1_pi`" function. I studied how you implemented the "exit button" and saw that you have the "`mui_LeaveForm(ui);`" function, so I used it to do what you proposed and I was successful. See the full code:

```
uint8_t mui_goto_or_back_form_w1_pi(mui_t *ui, uint8_t msg)
{
  if ( msg == MUIF_MSG_CURSOR_SELECT )
  {
    if ( mui_GetSelectableFieldTextOption(ui, ui->last_form_fds, ui->arg + ui->form_scroll_top) )
    {
      if ( (uint8_t)ui->text[0] == 254 ) // use MUI_255 for the return
      {
        mui_LeaveForm(ui);
        return 1;
      }
    }
  }
  return mui_u8g2_goto_form_w1_pi(ui,  msg);  // call the original MUIF
}
```

Another point: I tried to use "MUI_255", but when I use it, a problem occurs that when the display shows all the items in the list, they don't appear and I can't even interact with it.

As I said before, I'm still starting to study the library, so please comment if I've implemented something wrong or incorrectly. I thank you in advance for your help.

**Reply by olikraus** [reply, marked as answer]
You are right, 255 has a special meaning, we need to use 254 (or any other number between 1 and 254). `mui_LeaveForm` will not work, but yes, `mui_RestoreForm(ui);` is void.

The correct code is here: 
https://github.com/olikraus/u8g2/blob/e6a66ee0de9f863f898ddad5fe50a112d159de80/sys/sdl/mui_all/main.c#L213-L229

The muif table entry is here:
https://github.com/olikraus/u8g2/blob/e6a66ee0de9f863f898ddad5fe50a112d159de80/sys/sdl/mui_all/main.c#L305

and finally the menu is here:
https://github.com/olikraus/u8g2/blob/e6a66ee0de9f863f898ddad5fe50a112d159de80/sys/sdl/mui_all/main.c#L439-L455

**Reply by JosehL7** [reply]
Hi Olikraus,

Thank you very much for your help.
